packages/build-utils/src/file-ref.ts

```diff
@@ -139,3 +142,30 @@ export default class FileRef implements FileBase {
     });
   }
 }
+
+/**
+ * Get the value of the environment variable `key` as a valid URL string or `null` if unset.
+ *
+ * @param key The `process.env` member which holds the value to validate
+ * @returns The valid URL string or `null` if `key` is not set on `process.env`
+ */
+function getEnvAsUrlOrThrow(
+  key: keyof (typeof process)['env']
+): ReturnType<URL['toString']> | null {
+  const value = process.env[key];
+
+  if (value === undefined) return null;
+
+  try {
+    new URL(value);
```

Comment:
This constructor performs validation as a side effect. It can be swapped for `URL.canParse(value)` when we support >= Node.js 20.x as it was added in `19.9.0` ([ref](https://nodejs.org/docs/latest-v19.x/api/url.html#urlcanparseinput-base)).